src/SFML/Graphics/Font.cpp

```diff
@@ -603,43 +606,43 @@ Glyph Font::loadGlyph(Uint32 codePoint, unsigned int characterSize, bool bold, f
         glyph.bounds.height =  static_cast<float>(face->glyph->metrics.height)       / static_cast<float>(1 << 6) + outlineThickness * 2;
 
         // Extract the glyph's pixels from the bitmap
-        m_pixelBuffer.resize(width * height * 4, 255);
+        m_pixelBuffer.assign(width * height, 0x00FFFFFF);
         const Uint8* pixels = bitmap.buffer;
         if (bitmap.pixel_mode == FT_PIXEL_MODE_MONO)
         {
             // Pixels are 1 bit monochrome values
-            for (int y = 0; y < height; ++y)
+            for (int y = padding; y < height - padding; ++y)
             {
-                for (int x = 0; x < width; ++x)
+                for (int x = padding; x < width - padding; ++x)
                 {
                     // The color channels remain white, just fill the alpha channel
-                    std::size_t index = (x + y * width) * 4 + 3;
-                    m_pixelBuffer[index] = ((pixels[x / 8]) & (1 << (7 - (x % 8)))) ? 255 : 0;
+                    std::size_t index = x + y * width;
+                    m_pixelBuffer[index] = ((pixels[(x - padding) / 8]) & (1 << (7 - ((x - padding) % 8)))) ? 0xFFFFFFFF : 0x00FFFFFF;
                 }
                 pixels += bitmap.pitch;
             }
         }
         else
         {
             // Pixels are 8 bits gray levels
-            for (int y = 0; y < height; ++y)
+            for (int y = padding; y < height - padding; ++y)
             {
-                for (int x = 0; x < width; ++x)
+                for (int x = padding; x < width - padding; ++x)
                 {
                     // The color channels remain white, just fill the alpha channel
-                    std::size_t index = (x + y * width) * 4 + 3;
-                    m_pixelBuffer[index] = pixels[x];
+                    std::size_t index = x + y * width;
+                    m_pixelBuffer[index] = 0x00FFFFFF | (pixels[x - padding] << 24);
                 }
                 pixels += bitmap.pitch;
             }
         }
 
         // Write the pixels to the texture
-        unsigned int x = glyph.textureRect.left;
-        unsigned int y = glyph.textureRect.top;
-        unsigned int w = glyph.textureRect.width;
-        unsigned int h = glyph.textureRect.height;
-        page.texture.update(&m_pixelBuffer[0], w, h, x, y);
+        unsigned int x = glyph.textureRect.left - padding;
+        unsigned int y = glyph.textureRect.top - padding;
+        unsigned int w = glyph.textureRect.width + 2 * padding;
+        unsigned int h = glyph.textureRect.height + 2 * padding;
+        page.texture.update(reinterpret_cast<const Uint8*>(&m_pixelBuffer[0]), w, h, x, y);
```

Comment:
The idea was that because the unused area of m_pixelBuffer should contain transparent white pixels, it could be done once during initialization instead of every time a glyph is added. This scales better with bigger glyphs and the more glyphs are added.

The only reason I changed it to Uint32 is because that way initialization would only consist of a single line (the assign). Now that I think of it, since a single memset wouldn't be able to be used internally either (it can only assign a single byte value to a range), the assign would loop over each T anyway. I'll probably revert the storage back to Uint8 and loop initialize using a Uint8 array. If the optimizer isn't too dumb, it should end up producing pretty similar results.